### PR TITLE
New Plasmamen Helmets

### PR DESCRIPTION
**Pull request by DrMacCool**
## About The Pull Request
Due to their species traits, plasmamen have been completely unable to visually customise, being perpetually trapped in their samey envirosuits. This pull allows for plasmamen to break up the monotony with a selection of helmets. This pull specifically adds two new variants:-

<details>
<summary>Mark II</summary>

image

</details>
<details>
<summary>Protective</summary>

image

</details>

Credits:-
Sprites: DrMacCool
Code: Archanial

Prerence related defines have been moved to their file in defines.

New prereference option - plasmaman helmet. Your existing helmet will be deleted and you will be equipped with new one upon spawning.
This PR also bumps our savefile version to 36

## Why It's Good For The Game
Plasmamen finally get some aesthetic choice.

## Changelog
:cl:
add: Plasmamen can now choose from a selection of helmets.
/:cl:

### CODE REVIEW

File: code/modules/mob/dead/new_player/sprite_accessories.dm [hunk 1]
```diff
@@ -838,6 +838,10 @@
 	name = "Wisp"
 	icon_state = "hair_wisp"
 
+
+
+
+
```
Comment (Archanial) [inline code/modules/mob/dead/new_player/sprite_accessories.dm]:
```suggestion
```

**Comment by zeskorion** [comment]
You could make this a whole lot easier if you could figure out how to make the sprite greyscale, with an overlay for the stripes (also greyscale), and finally adding the visor and a any accessories as a third overlay

**Comment by Archanial** [comment]
> You could make this a whole lot easier if you could figure out how to make the sprite greyscale, with an overlay for the stripes (also greyscale), and finally adding the visor and a any accessories as a third overlay

this is true but gags are not merged yet

**Comment by Kapu1178** [comment]
I absolutely hate the way you chose to code this, Archanial. If you're going to make it this arbitrary (mutant features), just make it so you can use it in hand to change the icon state. This is just awful.

**Comment by ghost** [comment]
I have two problems with this, and similar PRs.

1) The enviro-helmet is a distinctive part of plasmamen, adding a bunch of fluff visual variants that only sort of look like envirohelmets, particularly the armored variants, bloats their visual design and makes them less distinguishable from other crewmembers. Not every player NEEDS to have cosmetic customization, especially when they choose to play as a species with deliberate restrictions on their equipment.

2) Adding too many cosmetic variants of outfits/uniforms/shells creates a confusing mess for players when it comes to identifying what the fuck is on their screen and codebases should pay more mind to preventing this. You don't need to look further than the bloated list of borg sprites to see the effect unnecessary cosmetic bloat has on gameplay.

**Comment by Kapu1178** [comment]
> I have two problems with this, and similar PRs.
> 
>     1. The enviro-helmet is a distinctive part of plasmamen, adding a bunch of fluff visual variants that only sort of look like envirohelmets, particularly the armored variants, bloats their visual design and makes them less distinguishable from other crewmembers. Not every player NEEDS to have cosmetic customization, especially when they choose to play as a species with deliberate restrictions on their equipment.
> 
>     2. Adding too many cosmetic variants of outfits/uniforms/shells creates a confusing mess for players when it comes to identifying what the fuck is on their screen and codebases should pay more mind to preventing this. You don't need to look further than the bloated list of borg sprites to see the effect unnecessary cosmetic bloat has on gameplay.

shift + click.

**Comment by DrMacCool** [comment]
> I have two problems with this, and similar PRs.
> 
>     1. The enviro-helmet is a distinctive part of plasmamen, adding a bunch of fluff visual variants that only sort of look like envirohelmets, particularly the armored variants, bloats their visual design and makes them less distinguishable from other crewmembers. Not every player NEEDS to have cosmetic customization, especially when they choose to play as a species with deliberate restrictions on their equipment.
> 
>     2. Adding too many cosmetic variants of outfits/uniforms/shells creates a confusing mess for players when it comes to identifying what the fuck is on their screen and codebases should pay more mind to preventing this. You don't need to look further than the bloated list of borg sprites to see the effect unnecessary cosmetic bloat has on gameplay.

I agree that the 'protective' variant pushes the limits of the design and I would be fine giving it the chop. However, I stand by the fact that the 'Mark II' remains aesthetically faithful to original envirohelmet design without pushing it in any kind of problematic direction. I understand your concern, but it would take a lot more than this to push the artstyle or the visual readability to the degree that those cyborg sprites do. Neither designs significantly deviate from the visual motifs of the original. You still have your purple skeletons in a job-coloured spacesuit.

The only bloat I see is the sheer quantity of sprites needed to make this work. But the only way to solve that is through a complete up-ending of how plasmamen clothing works. I'm nowhere near code-savvy enough to have an informed opinion outside of knowing it would be a big job, so please debate that with Arch instead.

File: code/modules/client/preferences_savefile.dm [hunk 2]
```diff
@@ -69,6 +69,8 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 		S.dir.Remove("overhead_chat")
 	if(current_version < 35)
 		see_balloon_alerts = BALLOON_ALERT_ALWAYS
+	if(current_version < 36)
+		WRITE_FILE(S["helmet_style"], HELMET_DEFAULT)
```
Comment (PowerfulBacon) [inline code/modules/client/preferences_savefile.dm]: Current version savefile updates is for variables where the default value has changed / the existing value is outdated, any new values added will be set to the default anyway.

File: code/modules/client/preferences_savefile.dm [hunk 3]
```diff
@@ -433,6 +436,9 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	if(!features["ethcolor"] || features["ethcolor"] == "#000")
 		features["ethcolor"] = GLOB.color_list_ethereal[pick(GLOB.color_list_ethereal)]
 
+	if(!helmet_style)
```
Comment (ike709) [inline code/modules/client/preferences_savefile.dm]: Also check that it's a valid option in the list of helmet options.

File: code/modules/client/preferences_savefile.dm (hunk 2 above)
Comment (ike709) [inline code/modules/client/preferences_savefile.dm]: Don't burn a savefile version, your sanity check should default it to `HELMET_DEFAULT` anyways

**Pull request merged**